### PR TITLE
json-c: run fuzz test from upstream repository

### PR DESCRIPTION
**Pull request by simonresch**
This commit removes the json-c fuzz test from the OSS-Fuzz repo in favor of running the [fuzz test that is maintained directly in the json-c repository](https://github.com/json-c/json-c/blob/master/fuzz/tokener_parse_ex_fuzzer.cc).

### CODE REVIEW

**Review by DavidKorczynski — COMMENTED** [review]
Could you assist with getting maintainer approval?

File: projects/json-c/build.sh
Comment (hawicz) [inline projects/json-c/build.sh:23]: What's the "fuzz/" directory, and why do we want to use it instead of $SRC?  I don't see it mentioned at eg https://github.com/google/oss-fuzz/blob/master/docs/getting-started/new_project_guide.md

Comment (simonresch) [reply projects/json-c/build.sh]: That is the [fuzz](https://github.com/json-c/json-c/tree/master/fuzz) directory inside the json-c repository. The build script is executed inside a checkout of json-c (see https://github.com/google/oss-fuzz/blob/master/projects/json-c/Dockerfile#L20). I can change this to `$SRC/json-c/fuzz` for a more explicit path.

Comment (simonresch) [reply projects/json-c/build.sh]: @hawicz Anything else I could add to clear up this PR?

If you want to try the changes out for yourself you can clone this branch via `git clone -b use-upstream-json-c-fuzz-test git@github.com:simonresch/oss-fuzz.git`. Then inside the `oss-fuzz` directory run `python3 infra/helper.py introspector json-c` to build all fuzz tests and briefly execute them. With the changes from this PR it should execute all fuzz tests from https://github.com/json-c/json-c/tree/master/fuzz instead of the single fuzz test that is currently configured in the oss-fuzz repository.

Comment (hawicz) [reply projects/json-c/build.sh]: lgtm

**Comment by simonresch** [comment]
@DavidKorczynski could you take a look at this PR? Maintainer approval is given [here](https://github.com/google/oss-fuzz/pull/12741#discussion_r1859211709).

**Review by DavidKorczynski — APPROVED** [review]

**Pull request merged**
